tests/algorithms/test_torchscript.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import torch
+from torchvision.models import resnet50
+
+from composer.functional import (apply_blurpool, apply_factorization, apply_ghost_batchnorm, apply_squeeze_excite,
+                                 apply_stochastic_depth)
+
+algo_kwargs = {apply_stochastic_depth: {'stochastic_method': 'block', 'target_layer_name': 'ResNetBottleneck'}}
+
+
+@pytest.fixture
+def input():
+    return torch.Tensor(1, 3, 224, 224)
+
+
+@pytest.mark.parametrize("surgery_method", [
+    pytest.param(apply_blurpool),
+    pytest.param(apply_factorization, marks=pytest.mark.xfail),
+    pytest.param(apply_ghost_batchnorm, marks=pytest.mark.xfail),
+    pytest.param(apply_squeeze_excite),
+    pytest.param(apply_stochastic_depth, marks=pytest.mark.xfail),
```

Comment:
Why does this fail for train and not for eval? Is the sample function that gets called during training not torch scriptable?

Reply:
Yes it seems difficult to torch script, in particular the `torch.Generator` type. I believe if that were removed, it may work.

Reply:
If this becomes high priority, happy to refactor since I've wanted to refactor this for so long